### PR TITLE
API: Fix prove bug on API

### PR DESCRIPTION
**Pull request by algoidan**
## Summary

This PR fixes a bug on algod's API. When a tree contains a missing child (not a full tree), the api handler omits this from the proof response and leads to a root mismatch 

## Test Plan
Add unit tests as well as convert the e2e to test this edge case.

### CODE REVIEW

File: test/e2e-go/features/transactions/proof_test.go [hunk 1]
```diff
@@ -120,6 +121,19 @@ func TestTxnMerkleProof(t *testing.T) {
...
+	s, _, err := blk.DecodeSignedTxn(blk.Payset[proofresp.Idx])
+	a.NoError(err)
+
+	if s.ID().String() != txid.String() {
+		t.Log(txid.String())
+		for i := 0; i < len(blk.Payset); i++ {
+			s, _, _ := blk.DecodeSignedTxn(blk.Payset[i])
+			t.Log(s.Txn.ID())
+		}
+		t.FailNow()
```
Comment (tsachiherman) [inline test/e2e-go/features/transactions/proof_test.go:134]: I think that the intent here is correct - but you should log more useful information that would help you to debug the issue further.
In particular, I'd suggest dumping the entire `confirmedTx` ( which includes the block number ), as well as the entire block.

Comment (algoidan) [reply test/e2e-go/features/transactions/proof_test.go:134]: thanks for the advice

**Review by tsachiherman — APPROVED** [review]
